torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -86,6 +86,8 @@ constexpr int64_t kBasicChannelPriority = 0;
 #if TENSORPIPE_HAS_CUDA_IPC_CHANNEL && defined(USE_CUDA_NOT_ROCM)
 constexpr int64_t kCudaIpcChannelPriority = 300;
 constexpr int64_t kCudaXthChannelPriority = 400;
+constexpr int64_t kCudaBasicChannelPriority = 500;
```

Comment:
Same here, shouldn't be guarded by `TENSORPIPE_HAS_CUDA_IPC_CHANNEL`.

Reply:
Oh and actually this should have a much lower priority, like 100. Otherwise we'll always prefer sending over TCP rather than doing IPC.

The final priority order should be: BASIC < GDR < IPC < XTH.